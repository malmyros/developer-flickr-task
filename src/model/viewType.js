module.exports = {
  LARGE: 'LARGE',
  THUMBNAIL: 'THUMBNAIL',
};
