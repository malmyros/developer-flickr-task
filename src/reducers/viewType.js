import { THUMBNAIL } from '../model/viewType';
import { CHANGE_VIEW_TYPE } from '../types';

const viewType = (state = THUMBNAIL, action) => {
  switch (action.type) {
    case CHANGE_VIEW_TYPE:
      return action.viewType;
    default:
      return state;
  }
};

export default viewType;
