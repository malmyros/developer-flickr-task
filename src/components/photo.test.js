import React from 'react';
import { shallow } from 'enzyme';
import Photo from './photo';
import { LARGE } from '../model/viewType';

describe('photo', () => {
  it('should render without crashing', () => {
    const component = shallow(<Photo id={'1'}
                                     farm={0}
                                     title={'photo 1'}
                                     viewType={LARGE}/>);
    expect(component).toMatchSnapshot();
  });
});
