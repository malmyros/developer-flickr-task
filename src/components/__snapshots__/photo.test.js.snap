// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photo should render without crashing 1`] = `
<div
  className="col-12"
>
  <div
    className="card mb-4 shadow-sm"
  >
    <img
      alt="photo 1"
      className="card-img-top"
      id="1"
      src="data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%22348%22%20height%3D%22225%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20348%20225%22%20preserveAspectRatio%3D%22none%22%3E%3Cdefs%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%23holder_165e1d92529%20text%20%7B%20fill%3A%23eceeef%3Bfont-weight%3Abold%3Bfont-family%3AArial%2C%20Helvetica%2C%20Open%20Sans%2C%20sans-serif%2C%20monospace%3Bfont-size%3A17pt%20%7D%20%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cg%20id%3D%22holder_165e1d92529%22%3E%3Crect%20width%3D%22348%22%20height%3D%22225%22%20fill%3D%22%2355595c%22%3E%3C%2Frect%3E%3Cg%3E%3Ctext%20x%3D%22116.71875%22%20y%3D%22120.15%22%3EThumbnail%3C%2Ftext%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E"
      style={
        Object {
          "display": "block",
          "filter": "blur(25px)",
          "height": "100%",
          "transition": "0.5s filter linear",
          "width": "100%",
        }
      }
    />
    <div
      className="card-body"
    >
      <div>
        <a
          href="https://www.flickr.com/photos/undefined/1"
          rel="noopener noreferrer"
          target="_blank"
          title="photo 1"
        >
          photo 1
        </a>
         by 
        <a
          href="https://www.flickr.com/photos/undefined/"
          rel="noopener noreferrer"
          target="_blank"
        />
      </div>
      <p
        className="card-text"
      />
      <div
        className="d-flex justify-content-between align-items-center"
      >
        <small
          className="text-muted"
        >
          <Tags
            tags={Array []}
          />
        </small>
      </div>
    </div>
  </div>
</div>
`;
